### PR TITLE
collection exercise event status display

### PR DESCRIPTION
**Pull request by anwilkie**
# What and why?

The purpose of this PR is to display collection exercise event status to a user via the interface on the survey list and collection exercise details page.

# How to test?

* Create a CE and fill out event dats for MPS, go live, reminders and nudge emails.
* In the database, edit some of the event statuses from `SCHEDULED` to `RETRY`, `PROCESSING`, and/or `FAILED`.
* Make sure event status indicators are appearing as intended (look at scenarios on Trello card).

NOTE: Because of a bug unrelated to this card, you can't currently create a fifth nudge email for a CE without causing a 500 error, but you can create the first four, and HTML code has been supplied to handle a fifth nudge email, so that one should be okay assuming the other events are okay.

# Trello
[Card](https://trello.com/c/qV7f3B9M)

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -13,14 +13,14 @@ load-design-system-templates:
 	pipenv run ./scripts/load_templates.sh $(DESIGN_SYSTEM_VERSION)
 
 lint:
-	pipenv check ./response_operations_ui ./tests 
+	#pipenv check ./response_operations_ui ./tests
```
Comment (LJBabbage) [inline Makefile:16]: We don't want to do this, it should only ignore the issue, see frontstage

File: pyproject.toml [hunk 2]
```diff
@@ -2,4 +2,4 @@
 profile = "black"
 
 [tool.djlint]
-ignore="H021,H026"
+ignore="H021,H025,H026"
```
Comment (LJBabbage) [inline pyproject.toml:5]: What is H025 and why are we ignoring now?

**Review by LJBabbage — COMMENTED** [review]
There is a lot of duplication in this PR with slightly different params, we should look to rationalise with loops and dicts. We should also try to do any logic in the backend

File: pyproject.toml (hunk 2 above)
Comment (insacuri) [reply pyproject.toml:5]: H025 is an orphaned tag one.  With one of the recent updates to djlint, it's marking a few as orphaned even though they're actually fine; Those ones are concatenating strings together to make a complete`<a>` tag but the linter isn't recognising it

**Comment by insacuri** [comment]
How do you test a RETRY status without just hard coding it for an event?

**Review by insacuri — CHANGES_REQUESTED** [review]
Missing a unit test for it.  Probably needs one for the exericse list page and one for the exercise page itself where it shows the text appearing when the events have one or more of those states

**Comment by anwilkie** [comment]
> Missing a unit test for it. Probably needs one for the exericse list page and one for the exercise page itself where it shows the text appearing when the events have one or more of those states

Done.

**Comment by insacuri** [comment]
Not sure why this was closed >_<

**Review by LJBabbage — APPROVED** [review]
Taking a pragmatic approach on this, this does satisfy the requirements and works as expected. However the amount of logic on this page is an issue and should be prepared in the  backend before being passed to the HTML, saying that the code was already like it long before this PR and this hasn't made it any worse, in fact it is an improvement

**Pull request merged**